src/lib/ITensorsExtensions/src/itensor.jl

```diff
@@ -58,18 +61,53 @@ function eigendecomp(A::ITensor, linds, rinds; ishermitian=false, kwargs...)
   D, U = eigen(A, linds, rinds; ishermitian, kwargs...)
   ul, ur = noncommonind(D, U), commonind(D, U)
   Ul = replaceinds(U, vcat(rinds, ur), vcat(linds, ul))
-
   return Ul, D, dag(U)
 end
 
-function map_eigvals(f::Function, A::ITensor, inds...; ishermitian=false, kwargs...)
+function map_eigvals(f::Function, A::ITensor, Linds, Rinds; kws...)
+  Linds = isa(Linds, Index) ? [Linds] : collect(Linds)
+  Rinds = isa(Rinds, Index) ? [Rinds] : collect(Rinds)
+
+  # <fermions>
+  fermionic_itensor =
+    ITensors.using_auto_fermion() && ITensors.has_fermionic_subspaces(inds(A))
+  if fermionic_itensor
+    # If fermionic, bring indices into i',j',..,dag(j),dag(i)
+    # ordering with Out indices coming before In indices
+    # Resulting tensor acts like a normal matrix (no extra signs
+    # when taking powers A^n)
+    if all(j->dir(j)==ITensors.Out, Linds) && all(j->dir(j)==ITensors.In, Rinds)
+      ordered_inds = [Linds..., reverse(Rinds)...]
+    elseif all(j->dir(j)==ITensors.Out, Rinds) && all(j->dir(j)==ITensors.In, Linds)
+      ordered_inds = [Rinds..., reverse(Linds)...]
+    else
+      error(
+        "For fermionic exp, Linds and Rinds must have same directions within each set. Got dir.(Linds)=",
+        dir.(Linds),
+        ", dir.(Rinds)=",
+        dir.(Rinds),
+      )
+    end
+    A = permute(A, ordered_inds)
+    # A^n now sign free, ok to temporarily disable fermion system
```

Comment:
I should have suggested this in the `exp` PR, but probably it would be better to split this logic off into a separate function for the sake of readability (and maybe so that the logic can be reused elsewhere).

Maybe this could be split off into a function `A = make_bosonic(A::ITensor, Linds, Rinds)`. In that design, I believe `ordered_inds` used down below could be determined from `inds(A)`.